bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/SalusApiException.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.rest;
+
+import java.io.Serial;
+
+/**
+ * @author Martin Grześlowski - Initial contribution
+ */
+@SuppressWarnings("SerializableHasSerializationMethods")
+public class SalusApiException extends RuntimeException {
```

Comment:
```suggestion
public class SalusApiException extends Exception {
```

Reply:
Why you want to use `Exception` instead a `REx`? Isn't it not recommended (+it ruins your lambdas)

Reply:
I know this. The question was why to use `Excaption` and not `RuntimeException`. Checked exceptions are in general pain in the ass (i.e. you cannot nicely use them in lambdas)

Reply:
- _A checked exception is caught at compile time whereas a runtime or unchecked exception is, as it states, at runtime._
- _A checked exception must be handled either by re-throwing or with a try catch block, whereas an unchecked isn’t required to be handled._
- _A runtime exception is a programming error and is fatal whereas a checked exception is an exception condition within your code’s logic and can be recovered or re-tried from._

Because of this it is a check in our generic review checklist: https://github.com/openhab/openhab-addons/wiki/Review-Checklist